modules/services/yabai/default.nix

```diff
@@ -0,0 +1,107 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.yabai;
+
+  toYabaiConfig = opts:
+    concatStringsSep "\n" (mapAttrsToList
+      (p: v: "yabai -m config ${p} ${toString v}") opts);
+
+in
+
+{
+  options = {
+    services.yabai = with types; {
+      enable = mkOption {
+        type = bool;
+        default = false;
+        description = "Whether to enable the yabai window manager.";
+      };
+
+      package = mkOption {
+        type = path;
+        default = pkgs.yabai;
+        description = "The yabai package to use.";
+      };
+
+      enableScriptingAddition = mkOption {
+        type = bool;
+        default = false;
+        description = ''
+          Whether to enable yabai's scripting-addition.
+          SIP must be disabled for this to work.
+        '';
+      };
+
+      config = mkOption {
+        type = attrs;
+        default = {};
+        example = literalExample ''
+          {
+            focus_follows_mouse = "autoraise";
+            mouse_follows_focus = "off";
+            window_placement    = "second_child";
+            window_opacity      = "off";
+            top_padding         = 36;
+            bottom_padding      = 10;
+            left_padding        = 10;
+            right_padding       = 10;
+            window_gap          = 10;
+          }
+        '';
+        description = ''
+          Key/Value pairs to pass to yabai's 'config' domain, via the configuration file.
+        '';
+      };
+
+      extraConfig = mkOption {
+        type = str;
+        default = "";
+        example = literalExample ''
+          yabai -m rule --add app='System Preferences' manage=off
+        '';
+        description = "Extra arbitrary configuration to append to the configuration file";
+      };
+    };
+  };
+
+  config = mkMerge [
+    (mkIf (cfg.enable) {
+      environment.systemPackages = [ cfg.package ];
```

Comment:
Any point in including this in systemPackages?

Reply:
Simply having the ‘yabai’ binary in the user’s PATH means they can arbitrarily send messages to the socket. A generally exercised privilege when having yabai installed

Reply:
It's also necessary for `yabai` to be in the PATH when the config file is evaluated (it's just a shell script)

Reply:
Actually nevermind, while the user probably doesn't need it other services like skhd do and that currently just uses this as context.